src/sentry/integrations/discord/integration.py

```diff
@@ -308,17 +308,19 @@ def dispatch(self, request, pipeline):
 
         pipeline.bind_state("guild_id", request.GET["guild_id"])
         pipeline.bind_state("code", request.GET["code"])
-        try:
-            raw_state = json.loads(request.GET["state"])
-            pipeline.bind_state("use_setup", raw_state.get("useSetup"))
-        except Exception as error:
-            logger.info(
-                "identity.discord.request-token",
-                extra={
-                    "state": request.GET("state"),
-                    "guild_id": request.GET["guild_id"],
-                    "code": request.GET["code"],
-                },
-            )
-            return pipeline.error(error)
+        has_state = "state" in request.GET
+        if has_state:
+            try:
+                raw_state = json.loads(request.GET["state"])
+                pipeline.bind_state("use_setup", raw_state.get("useSetup"))
+            except Exception as error:
+                logger.info(
+                    "identity.discord.request-token",
+                    extra={
+                        "has_state": has_state,
+                        "guild_id": request.GET["guild_id"],
+                        "code": request.GET["code"],
+                    },
+                )
+                return pipeline.error(error)
```

Comment:
@17hogeju if we get a JSON error we can log it but I think we should let the pipeline continue